### PR TITLE
Add PressureControl ROS server and client

### PR DESCRIPTION
**Pull request by 708yamaguchi**
# Server usage

Set `control_pressure` as true.

```
<launch>

  <include file="$(find kxr_controller)/launch/kxr_controller.launch">
    <arg name="urdf_path" value="$(find yamaguchi_7axis_arm_no_worm_with_vacuum_base_ver3)/urdf/yamaguchi_7axis_arm_no_worm_with_vacuum_base_ver3_color.urdf" />
    <arg name="servo_config_path" value="$(find yamaguchi_7axis_arm_no_worm_with_vacuum_base_ver3)/config/servo_config.yaml" />
    <arg name="publish_sensor" value="false" />
    <arg name="control_pressure" value="true" />
    <arg name="control_loop_rate" value="20" />
  </include>

</launch>
```

# Client usage

## keep pressure under -10[kPa]

Set `threshold` field as -10 and `release` field as false.

Shell
```
$ rostopic pub /kxr_fullbody_controller/pressure_control_interface/goal kxr_controller/PressureControlActionGoal "header:
  seq: 0
  stamp:
    secs: 0
    nsecs: 0
  frame_id: ''
goal_id:
  stamp:
    secs: 0
    nsecs: 0
  id: ''
goal:
  board_idx: 39
  threshold: -10.0
  release: false" 
```

Python
```
roscd kxr_controller/scripts
python3 interface.py
In [1]: ri.send_pressure_control(39, threshold=-10, release=False)
```

Euslisp
```
roscd kxreus/euslisp
roseus kxr-interface.l
(kxr-init)
(send *ri* :send-pressure-control :board-idx 39 :threshold -10 :release nil)
```

## Connect to atmosphere

Set `release` field as true.

Shell
```
$ rostopic pub /kxr_fullbody_controller/pressure_control_interface/goal kxr_controller/PressureControlActionGoal "header:
  seq: 0
  stamp:
    secs: 0
    nsecs: 0
  frame_id: ''
goal_id:
  stamp:
    secs: 0
    nsecs: 0
  id: ''
goal:
  board_idx: 39
  threshold: -10.0
  release: true" 
```

Python
```
roscd kxr_controller/scripts
python3 interface.py
In [1]: ri.send_pressure_control(39, threshold=-10, release=True)
```

Euslisp
```
roscd kxreus/euslisp
roseus kxr-interface.l
(kxr-init)
(send *ri* :send-pressure-control :board-idx 39 :threshold -10 :release t)
```

## Get current pressure

Shell
```
rostopic echo /kxr_fullbody_controller/pressure/39
data: 2.004485607147217
---                      
data: -0.006360102444887161
---                    
data: 0.9152774810791016
---                       
```

Python
```
roscd kxr_controller/scripts
python3 interface.py
In [1]: ri.read_pressure(39)
Out[1]: data: -16.13501739501953
```

Euslisp
```
roscd kxreus/euslisp
roseus kxr-interface.l
(kxr-init)
(send (send *ri* :read-pressure :board-idx 39) :data)
-11.8201
```

# elf files

v0.6.3 https://github.com/nakane11/control_boards/releases/tag/v0.6.3.elf
v0.6.4 https://github.com/708yamaguchi/rcb4/releases/tag/v0.6.4.elf

### CODE REVIEW

**Comment by 708yamaguchi** [comment]
This PR is extended version of https://github.com/iory/rcb4/pull/67.

**Comment by 708yamaguchi** [comment]
TODO:

- How to decide board id.

```
    def search_air_board_ids(self):
        air_boards = self.all_air_boards()
        # TODO: How to decide (id*2) or (id*2+1)
        return np.array([air_board.id*2+1 for air_board in air_boards])
```

- Can we use raw board id (board_idx) in PressureControl.action? Should ROS system care about the board id?
```
uint16 board_idx  # Do not use board id? (Currently, bus-connected air board is not supported)
```

**Comment by nakane11** [comment]
Regarding `search_air_board_ids()`, what about filtering `self.id_vector` using `j.board_revision` as is done in  `all_air_boards()`?
Maybe self.id_vector needs to be updated with  `read_jointbase_sensor_ids()`

**Comment by 708yamaguchi** [comment]
@nakane11 
Thank you for your review.
Does the last commit match your idea?

**Comment by nakane11** [comment]
@708yamaguchi 
Thank you, that's exactly right.

File: ros/kxreus/euslisp/kxr-interface.l [hunk 1]
```diff
@@ -65,7 +65,12 @@
                                     (format nil "/~A/stretch_interface" controller-name))
                                   kxr_controller::StretchAction
                                   :groupname groupname))
-   (dolist (action (list servo-on-off-client stretch-client))
+   (setq pressure-control-client (instance ros::simple-action-client :init
+                                  (if namespace (format nil "/~A/~A/pressure_control_interface" namespace controller-name)
+                                    (format nil "/~A/pressure_control_interface" controller-name))
+                                  kxr_controller::PressureControlAction
+                                  :groupname groupname))
+   (dolist (action (list servo-on-off-client stretch-client pressure-control-client))
```
Comment (iory) [inline ros/kxreus/euslisp/kxr-interface.l:73]: The pressure-control-client is not supported on the rcb4 board, so it is better not to check it.

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by iory** [comment]
LGTM

**Pull request merged**